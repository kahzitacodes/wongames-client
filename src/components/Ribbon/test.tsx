import { screen } from '@testing-library/react'

import { Ribbon } from '.'
import { renderWithTheme } from '@/utils/tests/helpers'

describe('<Ribbon />', () => {
  it('should render the Ribbon', () => {
    const { container } = renderWithTheme(<Ribbon>Best seller</Ribbon>)

    expect(screen.getByText(/best seller/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the Ribbon with accent color by default', () => {
    const { container } = renderWithTheme(<Ribbon>Best seller</Ribbon>)

    expect(screen.getByText(/best seller/i)).toHaveStyle({
      'background-color': '#2496FF'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the Ribbon with danger color when color is passed', () => {
    const { container } = renderWithTheme(
      <Ribbon color="red">Best seller</Ribbon>
    )

    expect(screen.getByText(/best seller/i)).toHaveStyle({
      'background-color': '#FB6E78'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the Ribbon default size', () => {
    const { container } = renderWithTheme(
      <Ribbon color="accent">Best seller</Ribbon>
    )

    expect(screen.getByText(/best seller/i)).toHaveStyle({
      height: '3.6rem',
      'font-size': '1.4rem'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
