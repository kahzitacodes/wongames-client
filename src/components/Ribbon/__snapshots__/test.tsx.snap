// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon /> should render the Ribbon 1`] = `
.c0 {
  top: 1.6rem;
  display: flex;
  align-items: center;
  color: #FAFAFA;
  font-weight: 600;
  width: fit-content;
  position: absolute;
  background-color: #2496FF;
  height: 3.6rem;
  font-size: 1.4rem;
  padding: 0 3.2rem;
  right: -2rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c0::before {
  border-left-color: #1177D6;
  border-top-color: #1177D6;
}

.c0::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

<div
  class="c0"
  color="accent"
>
  Best seller
</div>
`;

exports[`<Ribbon /> should render the Ribbon default size 1`] = `
.c0 {
  top: 1.6rem;
  display: flex;
  align-items: center;
  color: #FAFAFA;
  font-weight: 600;
  width: fit-content;
  position: absolute;
  background-color: #2496FF;
  height: 3.6rem;
  font-size: 1.4rem;
  padding: 0 3.2rem;
  right: -2rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c0::before {
  border-left-color: #1177D6;
  border-top-color: #1177D6;
}

.c0::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

<div
  class="c0"
  color="accent"
>
  Best seller
</div>
`;

exports[`<Ribbon /> should render the Ribbon with accent color by default 1`] = `
.c0 {
  top: 1.6rem;
  display: flex;
  align-items: center;
  color: #FAFAFA;
  font-weight: 600;
  width: fit-content;
  position: absolute;
  background-color: #2496FF;
  height: 3.6rem;
  font-size: 1.4rem;
  padding: 0 3.2rem;
  right: -2rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c0::before {
  border-left-color: #1177D6;
  border-top-color: #1177D6;
}

.c0::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

<div
  class="c0"
  color="accent"
>
  Best seller
</div>
`;

exports[`<Ribbon /> should render the Ribbon with danger color when color is passed 1`] = `
.c0 {
  top: 1.6rem;
  display: flex;
  align-items: center;
  color: #FAFAFA;
  font-weight: 600;
  width: fit-content;
  position: absolute;
  background-color: #FB6E78;
  height: 3.6rem;
  font-size: 1.4rem;
  padding: 0 3.2rem;
  right: -2rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c0::before {
  border-left-color: #F55D5D;
  border-top-color: #F55D5D;
}

.c0::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

<div
  class="c0"
  color="red"
>
  Best seller
</div>
`;
